pennylane/operation.py

```diff
@@ -506,20 +518,6 @@ def __repr__(self):
             return f"{self.name}({params}, wires={self.wires.tolist()})"
         return f"{self.name}(wires={self.wires.tolist()})"
 
-    @property
-    def num_params(self):
-        """Number of trainable parameters that this operator expects to be fed via the
-        dynamic `*params` argument.
-
-        By default, this property returns as many parameters as were used for the
-        operator creation. If the number of parameters for an operator subclass is fixed,
-        this property can be overwritten to return the fixed value.
-
-        Returns:
-            int: number of parameters
-        """
-        return self._num_params
```

Comment:
I am never too sure if one should define private attributes in `__init__` and use `@property` or if one should just make the attributes public. 

But since we use private ones for `wires`, `name` `id` etc, would it not be good to keep it consistent?

Reply:
Update: if it is true that a property always overwrites class attributes, then this makes sense!

Reply:
Yes, as far as I understand it does. I think my suggested implementation is essentially the only one that allows sub-classes to chose between implementing num_params as a class property, an instance property, or not at all.